main.go

```diff
@@ -455,6 +458,38 @@ func mainErr(args []string) error {
 	transformed := args[1:]
 	// log.Println(tool, transformed)
 	if transform != nil {
+		if len(args) == 2 && args[1] == "-V=full" {
+			cmd := exec.Command(args[0], args[1:]...)
+			out, err := cmd.Output()
+			if err != nil {
+				if err, _ := err.(*exec.ExitError); err != nil {
+					return fmt.Errorf("%v: %s", err, err.Stderr)
+				}
+				return err
+			}
+			line := string(bytes.TrimSpace(out))
+			f := strings.Fields(line)
+			if len(f) < 3 || f[0] != tool || f[1] != "version" || f[2] == "devel" && !strings.HasPrefix(f[len(f)-1], "buildID=") {
+				return fmt.Errorf("%s -V=full: unexpected output:\n\t%s", args[0], line)
+			}
+			var toolID []byte
+			if f[2] == "devel" {
+				// On the development branch, use the content ID part of the build ID.
+				toolID = decodeHash(contentID(f[len(f)-1]))
+			} else {
+				// For a release, the output is like: "compile version go1.9.1 X:framepointer".
+				// Use the whole line.
+				toolID = []byte(line)
+			}
+
+			out = bytes.TrimSpace(out) // no trailing newline
+			contentID, err := ownContentID(toolID)
+			if err != nil {
+				return fmt.Errorf("cannot obtain garble's own version: %v", err)
+			}
+			fmt.Printf("%s +garble buildID=_/_/_/%s\n", line, contentID)
```

Comment:
What are these underscores for? Are they to signify that we are ignoring certain parts of the buildID?

Reply:
Indeed. I'll add a comment about this.

Reply:
I had explained this in the commit message, so I just copied that paragraph as a comment.